scheduler/src/cook/rest/api.clj

```diff
@@ -954,12 +987,31 @@
                (not (contains? (get-gpu-models-on-pool (config/valid-gpu-models) pool-name) requested-gpu-model)))
       (throw (ex-info (str "The following GPU model is not supported: " requested-gpu-model) {})))))
 
+(defn validate-job-disk
+  "Validates that a job requesting disk is satisfying the following conditions:
+    - User can request size but not type; User can request both size and type; User cannot request only type and not size
+    - Requested type must be a valid type in config
+    - Requested size must be less than the max size in config"
+  [pool-name {:keys [disk]}]
+  (let [{disk-request :request disk-limit :limit requested-disk-type :type} disk
+        max-size (or (get-max-disk-size-on-pool (config/disk) pool-name) 10000000)
```

Comment:
Can we just reject the job submission in this case?

Reply:
yeah, config should reject that, forgot to update this portion to reflect that